drivers/spi/spi_ll_stm32.c

```diff
@@ -431,14 +463,34 @@ static void spi_stm32_complete(const struct device *dev, int status)
 {
 	const struct spi_stm32_config *cfg = dev->config;
 	SPI_TypeDef *spi = cfg->spi;
+
+	if (status == 0 && (LL_SPI_GetMode(spi) == LL_SPI_MODE_MASTER)) {
+#ifdef CONFIG_SPI_STM32_INTERRUPT
+		if (ll_func_spi_is_busy(spi)) {
+			return;
+		}
+#else /* CONFIG_SPI_STM32_INTERRUPT */
+		while (ll_func_spi_is_busy(spi)) {
+			/* NOP */
+		}
+#endif
```

Comment:
You're changing function behavior when fifo is not enabled.
It would deserve a proper justification and a dedicated commit if this is justified